app/controllers/api/messaging.php

```diff
@@ -1602,14 +1603,183 @@
+                    ->schedule();
+            } else {
+                $messaging->trigger();
+            }
+        }
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic($message, Response::MODEL_MESSAGE);
+    });
+
+App::post('/v1/messaging/messages/sms')
+    ->desc('Create an sms.')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'messages.create')
+    ->label('audits.resource', 'messages/{response.$id}')
+    ->label('scope', 'messages.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'createSMS')
+    ->label('sdk.description', '/docs/references/messaging/create-sms.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MESSAGE)
+    ->param('messageId', '', new CustomId(), 'Message ID. Choose a custom ID or generate a random ID with `ID.unique()`. Valid chars are a-z, A-Z, 0-9, period, hyphen, and underscore. Can\'t start with a special char. Max length is 36 chars.')
+    ->param('providerId', '', new UID(), 'SMS Provider ID.')
+    ->param('to', [], new ArrayList(new Text(Database::LENGTH_KEY)), 'List of Topic IDs or List of User IDs or List of Target IDs.')
+    ->param('description', '', new Text(256), 'Description for Message.', true)
+    ->param('content', '', new Text(64230), 'SMS Content.')
+    ->param('status', 'processing', new WhiteList(['draft', 'processing']), 'Message Status. Value must be either draft or processing.', true)
+    ->param('deliveryTime', null, new DatetimeValidator(requireDateInFuture: true), 'Delivery time for message in ISO 8601 format. DateTime value must be in future.', true)
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('messaging')
+    ->inject('response')
+    ->action(function (string $messageId, string $providerId, array $to, string $description, string $content, string $status, ?string $deliveryTime, Database $dbForProject, Document $project, Messaging $messaging, Response $response) {
+        $messageId = $messageId == 'unique()' ? ID::unique() : $messageId;
+
+        $provider = $dbForProject->getDocument('providers', $providerId);
+
+        if ($provider->isEmpty() || $provider->getAttribute('type') !== 'sms') {
+            throw new Exception(Exception::PROVIDER_NOT_FOUND);
+        }
+
+        $message = $dbForProject->createDocument('messages', new Document([
+            '$id' => $messageId,
+            'providerId' => $provider->getId(),
+            'providerInternalId' => $provider->getInternalId(),
+            'to' => $to,
+            'description' => $description,
+            'data' => [
+                'content' => $content,
+            ],
+            'status' => $status,
+            'search' => $messageId . ' ' . $description . ' ' . $providerId,
```

Comment:
Let's make sure we set deliveryTime here

Reply:
done this